services/webnn/ort/graph_builder_ort.cc

```
@@ -1031,92 +1031,90 @@ GraphBuilderOrt::AddDequantizeLinearOperation(
+        block_size = input_shape[i] / scale_shape[i];
+        axis = i;
+        blocked_axis_count++;
+        // TODO(https://github.com/shiyi9801/chromium/issues/135): Consider to
+        // add emulation to support multi-dimensions blockwise.
+        if (blocked_axis_count > 1) {
+          return NewNotSupportedError(
+              "For blocked dequantization scale has the same shape as the "
+              "input or except for one dimension in which blocking is "
+              "performed");
         }
       }
-      // The shape of scale is the same as the shape of input.
-      if (blocked_axis_count == 0) {
-        axis = 0;
-        block_size = 1;
-      }
+    }
 
-      // Currently, OpenVINO only supports axis == 0 when scale.size == 2.
-      // https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/dequantize_linear.cpp#L228.
-      if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
-              switches::kWebNNOrtUseOpenvino) &&
-          axis != 0) {
-        input_name = PrependTranspose(input_name, {1, 0});
-        scale_name = PrependTranspose(scale_name, {1, 0});
-        zero_point_name = PrependTranspose(zero_point_name, {1, 0});
-        axis = 0;
-        need_transpose = true;
-      }
-    } else {
-      // The proposal of requiring scale and zeroPoint to be the same rank as
-      // the input is under discussion-
-      // https://github.com/webmachinelearning/webnn/pull/805#discussion_r1919498405
-      return NewNotSupportedError(
-          "Currently, ONNX only supports per-tensor, per-axis and block-wise "
-          "dequantizeLinear");
+    // Currently, OpenVINO only supports axis == 0 when scale.size == 2.
+    // https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/dequantize_linear.cpp#L228.
+    if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
+            switches::kWebNNOrtUseOpenvino) &&
+        axis != 0) {
```

```suggestion
        axis == 1 && scale_shape.size() == 2) {
```

---

Or we can report the error earlier as, something like
```
if (OV EP is used) {
  if (scale rank is not 2) {
    throw error;
  } else if (axis == 1) {
    do transpose;
  }
}
```

---

> if (OV EP is used) {
  if (scale rank is not 2) {
    throw error;
  } else if (axis == 1) {
    do transpose;
  }
}

Sounds good. report the error earlier in new commit.